File: lib/core/providers/contracts/flight_provider.dart
import '../../../core/domain/models/offers/flight_offer.dart';

abstract interface class FlightProvider {
  String get providerId;
  String get providerName;

  Future<List<FlightOffer>> searchFlights({
    required String origin,
    required String destination,
    required DateTime departure,
    DateTime? returnDate,
    int? passengers,
  });
}

File: backend/src/common/providers/flight.provider.ts
import { ProviderResult } from './provider.result';

export interface FlightProvider {
  providerId: string;
  providerName: string;
  searchFlights(params: {
    origin: string;
    destination: string;
    departure: Date;
    returnDate?: Date;
    passengers?: number;
  }): Promise<ProviderResult<any[]>>;
}
